### ISSUE TITLE
Wrong GOROOT after updating to new version of Go and removing old one with Brew

### ISSUE THREAD

**Issue by vgheri** [joefitzgerald/go-plus#575]
### Prerequisites

* [x] Have you tried launching `atom .` from the terminal in your project's directory?
* [x] Have you verified the output from `go env` is correct? If it is, please include the output in this issue.
* [x] Have you updated Atom to the latest version?
* [ ] Have you tried using [Atom Beta](https://atom.io/beta), which can be run side-by-side with Atom Stable? Is the behavior the same, or different?
* [x] Have you updated your Atom packages to the latest versions?
* [x] Have you read the [FAQ](https://github.com/joefitzgerald/go-plus/wiki/FAQ)?
* [x] Have you searched [the issues](https://github.com/joefitzgerald/go-plus/issues?utf8=%E2%9C%93&q=is%3Aissue) to see if others are experiencing the same issue?

### Description

After updating go version to 1.8 using the official go OS X package (previously was using brew installed versions 1.7.3 and 1.7.5), go-plus cannot navigate to definition nor display autocomplete hints nor display signatures in status bar for built-in objects due to wrong GOROOT setting.
This is the error I get:
![image](https://cloud.githubusercontent.com/assets/1318563/23183635/038a9d12-f87d-11e6-8f54-55fb97ceeff9.png)

It's trying to use GOROOT /usr/local/Cellar/go/1.7.3/libexec/src/database/sql whilst 
go env output for GOROOT reports `GOROOT="/usr/local/go"`

If I search `/usr/local/Cellar/go/1.7.3/` in the Atom config folder, I find in the path 
`atom/packages/go-plus/spec/config/fixtures/go-151-darwin.json` the following json keys hardcoded
```
...
"GOROOT": "/usr/local/Cellar/go/1.7.3/libexec",
"GOTOOLDIR": "/usr/local/Cellar/go/1.7.3/libexec/pkg/tool/darwin_amd64",
...
```
I don't know if it's related at all, just reporting for completeness.

### Output from `atom -v && apm -v`

```
Valerios-MacBook-Pro:acl vgheri$ atom -v && apm -v
Atom    : 1.14.3
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
apm  1.15.3
npm  3.10.5
node 4.4.5 x64
python 2.7.10
git 2.10.1
```

### Output From `go env`

```
Valerios-MacBook-Pro:acl vgheri$ go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/vgheri/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/9v/4b6h7ftd5ldfqv7gr82mbj340000gn/T/go-build280650015=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```

### Steps to Reproduce

1. Install go v1.7.3 from brew
2. Update go version and remove v1.7.3
3. Start Atom and try to go to definition of a built-in function or symbol

### Expected Behavior

Expected go-plus to retrieve correct value for GOROOT and correctly go to definition, show autocomplete values for built-in objects.

### Actual Behavior

Autocomplete, go to definition, signature in status bar do not work.
Wrong GOROOT value is retrieved.

**Comment by joefitzgerald** [comment]
Thinking that perhaps you already had Atom running and didn't close it? Try closing and reopening atom.

**Comment by joefitzgerald** [comment]
Also, you may want to update all tools. There's a menu option, and a command in the command pallette for `golang:update-tools`.

**Comment by vgheri** [comment]
Did both. Uninstalled, reinstalled, erased .atom folder, updated all tools reinstalling from scratch.

**Comment by joefitzgerald** [comment]
The "hardcoded" goroot you described earlier are in go-plus' internal tests and have no impact on the operation of the package.

**Comment by joefitzgerald** [comment]
Can you try running `which -a go`? And then `ls -al` for each result?

**Comment by vgheri** [comment]
I thought so about the fixtures, but it was my only lead :/
Sure here it comes:
```
Valerios-MacBook-Pro:acl vgheri$ which -a go
/usr/local/go/bin/go

Valerios-MacBook-Pro:acl vgheri$ ls -al /usr/local/go/bin/go
-rwxr-xr-x  1 root  wheel  9985892 Feb 16 18:56 /usr/local/go/bin/go
```

**Comment by joefitzgerald** [comment]
What's the output of:

* `go version`
* `env`
* `go env`

...from the terminal?

**Comment by joefitzgerald** [comment]
Are you open to a screen sharing session to figure this out? If so, please join the `#go-plus` channel in one of these Slack teams, and then direct message me your email address:

[![Slack](https://img.shields.io/badge/atom_slack-%23go--plus-blue.svg?style=flat)](https://atom-slack.herokuapp.com) [![Slack](https://img.shields.io/badge/gophers_slack-%23go--plus-blue.svg?style=flat)](https://gophersinvite.herokuapp.com)

**Comment by vgheri** [comment]
go env is in my first post.
Others are:
```
Valerios-MacBook-Pro:acl vgheri$ go version
go version go1.8 darwin/amd64
Valerios-MacBook-Pro:acl vgheri$ env
TERM_PROGRAM=Apple_Terminal
SHELL=/bin/bash
TERM=xterm-256color
TMPDIR=/var/folders/9v/4b6h7ftd5ldfqv7gr82mbj340000gn/T/
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.Qfpq2AuqNT/Render
TERM_PROGRAM_VERSION=388
OLDPWD=/Users/vgheri
TERM_SESSION_ID=973665B9-9140-4E32-A273-7CD51A57AB2A
USER=vgheri
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.yqZKPGxo6C/Listeners
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/usr/local/go/bin:/usr/local/protoc/bin:/Users/vgheri/go/bin
PWD=/Users/vgheri/go/src/github.com/vpcorp/acl
XPC_FLAGS=0x0
HISTCONTROL=ignoreboth:erasedups
XPC_SERVICE_NAME=0
SHLVL=1
HOME=/Users/vgheri
LOGNAME=vgheri
LC_CTYPE=UTF-8
GOPATH=/Users/vgheri/go
SECURITYSESSIONID=186a7
_=/usr/bin/env
```

Sure will do, thanks a lot!!

**Comment by emil-nasso** [comment]
I'm not 100% that this is related but i'm running into some issues and i think it is because i upgraded to go 1.8.

At first autocomplete didn't work. When i wrote `fmt.Print` i didn't get any suggestions.

I upgraded all tools, restated atom (and the computer). That seems to be working now again.

Cmd + click doesn't work however.

I get this error message: 

![https://dl.dropboxusercontent.com/s/5105g33rpea2l0v/2017-02-21%20at%2022.34.png](https://dl.dropboxusercontent.com/s/5105g33rpea2l0v/2017-02-21%20at%2022.34.png)

**Comment by vgheri** [comment]
Hello, just a quick reminder that I am available (even on demand) to try and understand what's wrong with my conf/the plugin/? .
Thanks again!

Valerio

**Comment by zmb3** [comment]
@vgheri - It sounds to me like you had installed the official binary packages from golang.org (which install to `/usr/local/go`, and also installed Go via brew, and somehow are seeing artifacts from both.

I would try completely removing Go (both of them):
- `brew uninstall go`
- `sudo rm -rf /usr/local/go`
- `sudo rm /etc/paths.d/go`

And then reinstall via brew:
 - `brew install go`

**Comment by vgheri** [comment]
Thanks @zmb3 , I will try this procedure over the weekend.
Nevertheless it would be interesting to know why go-plus/atom/I don't know what is looking for a non existent path and if it's hardcoded somewhere.

**Comment by joefitzgerald** [comment]
go-plus just looks in the path. Then it looks in known locations. Add `/use/local/go/bin` to your path and it will begin working.

**Comment by joefitzgerald** [comment]
Acrually, sorry I see it is already in your path. Sounds like it may be worth trying @zmb3' procedure.

**Comment by vgheri** [comment]
Quick update: the problem disappeared completely on its own after a few days. 
I did nothing, strictly nothing.
To say that I'am puzzled is a bit of an understatement 😕

**Comment by joefitzgerald** [comment]
Sounds like a PATH issue. Did you reboot during those few days?

**Comment by vgheri** [comment]
Possibly.
But I had already rebooted several times before opening this issue.

**Comment by jimmiebtlr** [comment]
Sorry to raise the dead, ran into the same issue I think and came to this thread without an answer.

In my case I was trying to run mockery, but it was looking in the wrong version folder for stdlib packages.   The solution in my case was to re-install mockery.  I have no idea why that's necessary. 

If someone else runs into this, I'd suggest re-installing whatever lib your trying to use, or re-building any bins that you're trying to use.

**Comment by zmb3** [comment]
tl;dr - don't set GOROOT :-)

**Comment by jimmiebtlr** [comment]
Go root wasn't set on mine, though looks like it is in the above example.
